.github/workflows/build-pipeline.yml

```diff
@@ -0,0 +1,68 @@
+name: Build
+on: 
+  push:
+    branches: 
+      - main
+    paths-ignore:
+      - "README.md"
+  pull_request:
+    paths-ignore:
+      - "README.md"
+
+
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Build Docker
+      run: docker build --target test --tag todo-app:test .
+      
+    - name: Run tests in Docker
+      run: docker run --mount "type=bind,source=$(pwd),target=/usr/src/app" todo-app:test
+
+    - name: Check if Slack webhook URL secret exists
+      id: slack-secret-check
+      shell: bash
+      if: always()
+      run: |
+        if [ "${{ secrets.SLACK_WEBHOOK_URL }}" != '' ]; then
+          echo "available=true" >> $GITHUB_OUTPUT;
+        else
+          echo "available=false" >> $GITHUB_OUTPUT;
+        fi
```

Comment:
You can't directly access secrets in if statements. I tried using `env` but it didn't quite work... Is there a nicer way of doing this check? This is just something I got of SO

And if not, can I make functions in GitHub actions? I'm using a very similar one for the snyk job, just has different Id and checks different secret

Reply:
Fun question - I had a little play in [this repo](https://github.com/JackMead/github-actions-demo/blob/main/.github/workflows/example.yml) and I think that's meeting your question - we can either determine whether or not to run the job (as we might want to with e.g. Snyk) or just a single step (as with these steps) by converting to environment variables and referencing those:
```
jobs:
    jobConditional:
        env:
            SECRET_EXISTS: ${{ secrets.SECRET_EXISTS }}
        runs-on: ubuntu-latest
        if: ${{ github.env.SECRET_EXISTS != '' }}
        steps:
            - run: echo "This ran"
    stepConditional:
        runs-on: ubuntu-latest
        env:
            SECRET_EXISTS: ${{ secrets.SECRET_EXISTS }}
        steps:
            - run: echo "This ran"
              if: ${{ github.env.SECRET_EXISTS != '' }}
```

Reply:
Think I tried something similar when I said I tried using `env`, tried this suggestions and it still seems to be skipping over so might need to live with the current solution